nexus/src/authz/api_resources.rs

```diff
@@ -19,13 +19,87 @@
 // pass them to `is_allowed()`.  Using newtypes is a way to capture just the
 // parts we need for authorization.
 
-use super::actor::AuthenticatedActor;
-use super::roles::AuthzApiResource;
-use super::roles::AuthzResource;
+use super::actor::AnyActor;
+use super::context::Authorize;
+use super::roles::{
+    load_roles_for_resource, load_roles_for_resource_tree, RoleSet,
+};
+use super::Action;
+use super::{actor::AuthenticatedActor, Authz};
+use crate::authn;
+use crate::context::OpContext;
 use crate::db::fixed_data::FLEET_ID;
-use omicron_common::api::external::ResourceType;
+use crate::db::DataStore;
+use futures::future::BoxFuture;
+use futures::FutureExt;
+use omicron_common::api::external::{Error, LookupType, ResourceType};
 use uuid::Uuid;
 
+/// Describes an authz resource that corresponds to an API resource that has a
+/// corresponding ResourceType and is stored in the database
+///
+/// This is a helper trait used to impl [`AuthzResource`].
```

Comment:
Yeah, I noticed this too when I was applying your other feedback.  Fixed in 484ae6bcf0901fa55cd6c5142edb40ece4bca9e1.  I also tried a `cargo doc` run since this should emit warnings.  I believe there are no more warnings related to stuff in this change.  But there's a lot of warning noise in `cargo doc`, unfortunately.